### PR TITLE
Added optional setting http_proxy to push_image

### PR DESCRIPTION
**Pull request by xNUTx**
Original discussion here: https://github.com/HASwitchPlate/openHASP-custom-component/pull/117

I had it in 'main' instead of a separate branch, so it got a little confusing making patches for other things...

Functional it has no changes compared to the original PR.

> This will allow the user to set up an external reverse proxy to have the HA instance operate on HTTPS and push_image use that reverse proxy to avoid using HTTPS traffic processing on the plate.
> 
> Besides now working around a bug in the libraries that kills SSL traffic on the plates, it will also allow the plate not to deal with SSL encrypted connections at all (even when it is working again) to save processing power.

HA Automation Action example:

```
data:
  obj: p12b1
  image: >-
    https://myhainstance:portnumber{{
    state_attr('image.doorbell_event_image','entity_picture') }}
  http_proxy: http://internaladdress:portnumber
  width: 480
  height: 320
  fitscreen: 1
target:
  entity_id: openhasp.displayname
enabled: true
action: openhasp.push_image
```

### CODE REVIEW

File: custom_components/openhasp/__init__.py [hunk 1]
```diff
@@ -153,6 +154,7 @@ def hasp_object(value):
     {
         vol.Required(ATTR_IMAGE): vol.Any(cv.url, cv.isfile),
         vol.Required(ATTR_OBJECT): hasp_object,
+        vol.Optional(ATTR_PROXY): vol.Any(cv.url),
```
Comment (dgomes) [inline custom_components/openhasp/__init__.py:157]:
```suggestion
        vol.Optional(ATTR_PROXY): cv.url,
```

**Review by dgomes — CHANGES_REQUESTED** [review]

**Review by dgomes — APPROVED** [review]
Still don't see the point as discussed in https://github.com/HASwitchPlate/openHASP-custom-component/pull/117 but technically is OK

**Comment by xNUTx** [comment]
> Still don't see the point as discussed in #117 but technically is OK

If you are really that heartfelt about the naming, feel free to modify it as you see fit. Functionality is what matters, naming is trivial imho.

**Comment by xNUTx** [comment]
> So,can this be merged or not?

From my point of view, yes.

Technically it can be according to @dgomes.
If he insists on changing the name of the config variables he can choose to do so, I don't care about the naming that much and I personally need this feature in the HA component for it to work properly in my case and I am getting tired of making modifications every time there is an update on HA or it's components overwriting my required changes.

**Comment by fvanroie** [comment]
I'm not sure I understand the issue myself. I'm preparing for the HA 2024.9.0 release, so it would be a good time to merge.

**Comment by dgomes** [comment]
just rename it to what it is: `expose_on_address`

**Comment by xNUTx** [comment]
In my case though, the original image is hosted by HA on https, I actually have a reverse proxy in use to allow http access to the image url...

But again, I don't care anymore. I just need this feature, regardless of naming.

⁣--

Met vriendelijke groeten/Regards,

Barry Schut
a.k.a. [NUT]​

Op 30 aug. 2024 17:16, om 17:16, fvanroie ***@***.***> schreef:

**Comment by fvanroie** [comment]
It seems we are running in circles here...

**Pull request merged**